### PR TITLE
Add oVirt ovirt_host_pm module

### PR DESCRIPTION
**Pull request by machacekondra**
##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

ovirt_host_pm
##### SUMMARY

Add _ovirt_host_pm_ module to manage power management and fence agents of the oVirt hosts.

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina

File: cloud/ovirt/ovirt_host_pm.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    name:
+        description:
+            - "Name of the the host to manage."
+        required: true
+        aliases: ['host']
+    state:
+        description:
+            - "Should the host be present/absent/started/stopped/restarted"
+        choices: ['present', 'absent', 'started', 'stopped', 'restarted']
+        default: present
```
Comment (mwperina) [inline cloud/ovirt/ovirt_host_pm.py:49]: Why do we need to change host status when updating fence agents? I'd prefer to update fence agents without host status change

File: cloud/ovirt/ovirt_host_pm.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+    username:
+        description:
+            - "Username to be used to connect to power management interface."
+    password:
+        description:
+            - "Password of the user specified in C(username) parameter."
+    type:
+        description:
+            - "Type of the power management. oVirt predefined values are I(drac5), I(ilo), I(ipmilan), I(rsa),
+               I(bladecenter), I(alom), I(apc), I(eps), I(wti), I(rsb), but user can have defined custom type."
```
Comment (mwperina) [inline cloud/ovirt/ovirt_host_pm.py]: Please add all fence agents supported in 4.0:

apc, apc_snmp, bladecenter, cisco_ucs, drac5, drac7 ,eps, hpblade, ilo, ilo2, ilo3, ilo4, ilo_ssh, ipmilan, rsa, rsb, wti

**Review by mwperina — CHANGES_REQUESTED** [review]

File: cloud/ovirt/ovirt_host_pm.py (hunk 1 above)
Comment (machacekondra) [reply cloud/ovirt/ovirt_host_pm.py:49]: This is incorrect of course, I will remove it, thanks.

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! :+1:

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by machacekondra** [comment]
shipit

**Comment by gregdek** [comment]
Thanks @machacekondra. Since you are one of the maintainers (@machacekondra) of this module and you commented with "shipit", we are marking this PR for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks again to @machacekondra. This module is going into community review.

We notice that you are one of the maintainers (@machacekondra) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/ovirt/ovirt_host_pm.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+try:
+    import ovirtsdk4 as sdk
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
```
Comment (ryansb) [inline cloud/ovirt/ovirt_host_pm.py:26]: Please check this in the module code and raise an error instead of having a NameError later in the script.

Comment (mwperina) [reply cloud/ovirt/ovirt_host_pm.py:26]: @ryansb 
Ryan, please take a look at check_sdk() function on line 184, that's where we handle user friendly error if required SDK and its version is not present. Having this check on on place inside utils module seems to us like much better solution than do the same check in each module. Not to mention that when we will need to raise required SDK version we will do that on place

**Pull request merged**